### DISCUSSION TITLE
Use MD/MDd runtimes for msvc by default

### DISCUSSION THREAD

**Discussion by waruqi** [xmake-io/xmake#5526] [Polls]
The current version of xmake does not add any msvc flags by default, and the compiler will use the MT/MTd runtime by default.

In a future xmake 3.0 release, would you consider changing to use the MD/MDd runtime by default?

**Comment by SirLynix** [comment]
I vote to use MD/MDd by default as this is the default for Visual Studio generated projects (it's also safer than MT when using .dll)

**Comment by qudix** [comment]
Best to keep in mind this problem: https://developercommunity.visualstudio.com/t/Access-violation-with-std::mutex::lock-a/10664660#T-N10668856

> if you’re seeing crashes due to null dereferences in mutex locking machinery, you’re deploying a program built with new STL headers, but without a sufficiently new msvcp140.dll, which is unsupported. You need to be (re)distributing a new STL DLL too. (This is what’s causing the “not all users see this” behavior - if a VS 2022 17.10 VCRedist has been independently installed on the machine - which will be the case for your developer machine - then everything will happen to work.)

**Comment by waruqi** [comment]
you can try this patch. https://github.com/xmake-io/xmake/pull/5626

```lua
set_policy("compatibility.version", "3.0")
```
